### PR TITLE
[SPL] Regression testing options

### PR DESCRIPTION
**Pull request by watt**
- expose layout result for "snapshot" comparison
- hoist SwiftUI-SPL layout options up to the enum
- fix strict stack layout in some cases I found in regression tests

### CODE REVIEW

File: BlueprintUI/Sources/Element/ElementContent.swift [hunk 1]
```diff
@@ -48,17 +48,14 @@ public struct ElementContent {
...
+        case .singlePass(let options):
             return storage.sizeThatFits(
                 proposal: constraint,
                 context: .init(
                     // TODO: Hoist upward?
                     cache: SPCacheNode(
                         path: "m",
-                        options: SPCacheOptions(
-                            hintRangeBoundaries: true,
-                            searchUnconstrainedKeys: false
```
Comment (n8chur) [inline BlueprintUI/Sources/Element/ElementContent.swift:60]: Are we sure we want to use a valid of `false` for `searchCnconstrainedKeys` here? Do we have a TODO list item to dig deeper into the performance trade-off here?

File: BlueprintUI/Sources/Layout/Stack.swift [hunk 2]
```diff
@@ -1012,22 +1012,17 @@ extension StackLayout {
...
-        }()
-        
-        switch (alignment, crossPressure) {
-        case (.fill, .natural):
+        // If we're in fill mode, we'll do an additional pass to ensure we fill the assigned space,
+        // in case:
+        // - cross pressure is natural, so we still need to fill the calculated size
+        // - we skipped a pass during cross segments, and didn't fill along axis yet
+        switch alignment {
+        case .fill:
```
Comment (n8chur) [inline BlueprintUI/Sources/Layout/Stack.swift:1025]: oof, this is another hit to Strict SPL performance, right?

**Review by n8chur — APPROVED** [review]

File: BlueprintUI/Sources/Layout/LayoutMode.swift [hunk 3]
```diff
@@ -4,6 +4,6 @@ public enum LayoutMode: Equatable {
     public static let `default`: Self = .strictSinglePass
 
     case standard
-    case singlePass
+    case singlePass(options: SPCacheOptions = .default)
     case strictSinglePass
```
Comment (n8chur) [inline BlueprintUI/Sources/Layout/LayoutMode.swift:8]: Feel free to discard if you don't like this pattern:

```suggestion
    case strictSinglePass
    
    public static let singlePass: Self = .singlePass()
```

File: BlueprintUI/Sources/Element/ElementContent.swift (hunk 1 above)
Comment (watt) [reply BlueprintUI/Sources/Element/ElementContent.swift:60]: Hm, I thought you had convinced me previously that it didn't matter during measurement, but I could be misremembering. Anyhow, I added an item to come back to this.

File: BlueprintUI/Sources/Layout/Stack.swift (hunk 2 above)
Comment (watt) [reply BlueprintUI/Sources/Layout/Stack.swift:1025]: Yeah. There might be a tighter condition we could apply, but this much fixed the regression tests.

File: BlueprintUI/Sources/Element/ElementContent.swift [hunk 4]
```diff
@@ -48,17 +48,14 @@ public struct ElementContent {
...
         case .standard:
             return storage.measure(in: constraint, environment: environment, cache: cache)
-        case .singlePass:
+        case .singlePass(let options):
             return storage.sizeThatFits(
                 proposal: constraint,
                 context: .init(
                     // TODO: Hoist upward?
                     cache: SPCacheNode(
                         path: "m",
```
[the remark this answers is not in the document]
Comment (watt) [reply BlueprintUI/Sources/Element/ElementContent.swift:57]: > For my own education, what does "m" refer to here?

"measure". A little (temporary) trick to make it easier to distinguish out of band measurements during layout in Instruments.

**Pull request merged**